model_read_only/migrations/14.0.2.0.0/post-migration.py

```diff
@@ -0,0 +1,31 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        """
+            SELECT ir_model_id, res_groups_id
+            FROM ir_model_group_readonly_rel_tmp;
+        """,
+    )
+    data = env.cr.fetchall()
+    grouped = {}
+    for item in data:
+        key = item[0]
+        value = item[1]
+        if key in grouped:
+            grouped[key].append(value)
+        else:
+            grouped[key] = [value]
+    for k, v in grouped.items():
+        env["model.readonly.restriction"].create(
```

Comment:
Next, instead of do `create` in cycle, you can prepare values for `create` function and then do `create`.
For example:
```py
vals_list = [{"model_id": model_id, "group_ids": group_ids} for model_id, group_ids in grouped]
env["model.readonly.restriction"].create(vals_list)
```